modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/objects/fields/MultiselectPicklist.tsx

```diff
@@ -0,0 +1,190 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayMultiSelect from '@clayui/multi-select';
+import {Item} from '@clayui/multi-select/lib/types';
+import classnames from 'classnames';
+import {fetch} from 'frontend-js-web';
+import React, {useCallback, useEffect, useState} from 'react';
+
+// @ts-ignore
+
+import {getListTypeEntries} from '../../data/listType';
+import ErrorMessage from '../ErrorMessage';
+import {TGenericFieldProps} from '../FieldsWrapper';
+
+type PartialListTypeEntry = {
+	key: string;
+	name_i18n: {
+		[localeCode: string]: string;
+	};
+};
+
+type MultiselectPicklistItem = {
+	label: string;
+	key: string;
+	value: string;
+};
+
+const MULTISELECT_PICKLIST_SEPARATOR = ', ';
+
+const _itemsToValue = (items: MultiselectPicklistItem[], i18n = false) =>
+	items.reduce(
+		(
+			valueAsString: string,
+			{key, label}: MultiselectPicklistItem
+		): string => {
+			const stringChunk = i18n ? label : key;
+
+			return valueAsString
+				? `${valueAsString}${MULTISELECT_PICKLIST_SEPARATOR}${stringChunk}`
+				: stringChunk;
+		},
+		''
+	);
+
+const _valueToItems = (
+	value: string | undefined,
+	sourceItems: MultiselectPicklistItem[]
+): MultiselectPicklistItem[] =>
+	sourceItems.filter((sourceItem) => value?.includes(sourceItem.key));
```

Comment:
True, I simply didn't like splitting and double-looping through 2 arrays, but forgot to reintroduce the separator to avoid cases like yours.

```javascript
const _valueToItems = (
	value: string | undefined,
	sourceItems: TMultiselectPicklistItem[]
): TMultiselectPicklistItem[] =>
	sourceItems.filter((sourceItem) =>
		`${value}${MULTISELECT_PICKLIST_SEPARATOR}`?.includes(
			`${sourceItem.key}${MULTISELECT_PICKLIST_SEPARATOR}`));
```
This will compare on the fly
```
`"key11, key2, key3 ,"` against, say, `"key1, "`, `"key11, "`, ...

Do you think we're sacrificing readability?